### PR TITLE
Fix #4263: New ContentVisibilitySelect is missing a CreatePublicConte…

### LINKED ISSUE #4263: New ContentVisibilitySelect is missing a CreatePublicContent permission check
The ContentVisibilitySelect form widget introduced with 1.6 does not check for CreatePublicContent permission on the container.

### PR DESCRIPTION
**Pull request by luke-**
…nt permission check

**What kind of change does this PR introduce?** (check at least one)

- [X] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [ ] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [ ] All tests are passing
- [ ] New/updated tests are included
- [X] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: protected/humhub/modules/ui/form/widgets/ContentVisibilitySelect.php [hunk 1]
```diff
@@ -41,7 +42,7 @@
 class ContentVisibilitySelect extends InputWidget
 {
     /**
-     * @var bool Automatically hide the field when no mulitple visibility modes available
+     * @var bool Automatically hide the field when no only one visibility option is available
```
Comment (ArchBlood) [inline protected/humhub/modules/ui/form/widgets/ContentVisibilitySelect.php:45]: Probably could change this to something simpler
`Hide the field when only one visibility option is available` or
`Automatically hide the field when only one visibility option is available`

Comment (luke-) [reply protected/humhub/modules/ui/form/widgets/ContentVisibilitySelect.php]: Ok, good point!

**Comment by buddh4** [comment]
When using this we need to be aware that this field may be not be set, could this have any sideeffects, perhaps there we should suggest setting a default value? Also we should add some default validation within the content layer or force a default visibility in `Content::beforeSave` in such cases?

Currently ther only is:

```php
if (!$this->visibility) {
   $this->visibility = self::VISIBILITY_PRIVATE;
}
```

**Comment by luke-** [comment]
@buddh4 I'm not sure I get your point. We could always render at least one hidden input. This should avoid empty values. Additionally, a validation in the model must of course take place.

**Comment by buddh4** [comment]
My point is, that each form model which loads the visibility value needs to manually implement a validation of the value against the same visibility rules implemented in this widget, which leads to redundancy. Therefore there should be a validator or helper in the core. And we should either block invalid visibility settings or overwrite the visibility if needed on content class level to avoid inconsistent data as for example public content on private spaces.

**Comment by luke-** [comment]
Okay, that makes sense. Then we drop the check/validation in the form field until we have a validator that we can use in both the model and the form field.

**Pull request closed without merging**